lib/src/data/user_profile.dart

```diff
@@ -0,0 +1,75 @@
+import "course.dart";
+import "utils.dart";
+
+/// Represents information about a normal user.
+class UserProfile {
+  /// The user's name.
+  final String name;
+
+  /// The user's current courses.
+  final List<Course> currentCourses;
+
+  /// The user's past courses.
+  final List<Course> pastCourses;
```

Comment:
Actually, better to just store the course IDs. Think about database duplication and what happens if a course changes

```suggestion
  /// The user's current courses.
  final Set<CourseID> currentCourses;

  /// The user's past courses.
  final Set<CourseID> pastCourses;
```